op-challenger/challenger/log_store_test.go

```diff
@@ -0,0 +1,59 @@
+package challenger
+
+import (
+	"testing"
+	// "time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestLogStore_NewLogStore tests the NewLogStore method on a [logStore].
+func TestLogStore_NewLogStore(t *testing.T) {
+	query := ethereum.FilterQuery{}
+	logStore := NewLogStore(query)
+	require.Equal(t, query, logStore.query)
+	require.Equal(t, []types.Log{}, logStore.logList)
+	require.Equal(t, make(map[common.Hash]types.Log), logStore.logMap)
+	require.Equal(t, SubscriptionId(0), logStore.currentSubId)
+	require.Equal(t, make(map[SubscriptionId]Subscription), logStore.subMap)
```

Comment:
These tests are probably too tied to the specifics of the implementation rather than testing required functionality. We want the tests to focus on the externally visible behaviour and be as decoupled as possible from the implementation.  The key behaviours of the log store would be:
* `Spawn` creates a subscription and adds any logs received from it to a list of logs
* When an subscription reports an error, it gets resubscribed.
* `Quit` exits all the go routines listening to subscriptions (not sure how to test this...)

The points of visibility we have are the mock client we pass in and the logs that come back out.  So for reading the logs I think we wind up with a test like:
```Golang
func TestReadLogsFromSubscription(t *testing.T) {
	query := ethereum.FilterQuery{}
	logStore := NewLogStore(query)
	// TODO: Pass client into NewLogStore instead of setting internal variable directly.
	client := &mockLogFilterClient{}
	logStore.client = client

	// Create the new subscription.
	err := logStore.Spawn()
	require.NoError(t, err)
	sub := client.LastSubscription()
	require.NotNil(t, sub)
	l := types.Log{BlockHash: common.Hash{0xaa}}
	sub.logs <- l

	timeout, tCancel := context.WithTimeout(context.Background(), 30*time.Second)
	defer tCancel()
	err = e2eutils.WaitFor(timeout, 500*time.Millisecond, func() (bool, error) {
		result, ok := logStore.GetLogByBlockHash(l.BlockHash)
		return ok && result.BlockHash == l.BlockHash, nil
	})
	require.NoError(t, err)
}
```

https://github.com/ethereum-optimism/optimism/commit/9a91c8e4d6022ec1eb196b01104d0b93d794c66a has all the changes I made to support that - the mock client needs to track the subscription calls for example.

I don't really like the fact that the waiter functions are in `e2eutils` - they're useful for any multithreaded test so possibly should be moved out of op-e2e but I'm not really sure where they should live currently.

Reply:
Attempted to clean up the tests. Will re-request your review here since a lot has changed if that's ok :)